icicle-source/src/Icicle/Source/ToCore/Exp.hs

```diff
@@ -108,17 +107,36 @@ convertExpQ q
  $ case contexts q of
     []
      -> convertExp $ final q
-    (Let _ b d:cs)
+    (Let _ PatDefault _:cs)
+     ->     -- A pattern with no binding doesn't need to inspect its arguments
+            convertExpQ $ Query cs $ final q
+    (Let _ (PatVariable b) d:cs)
      -> do  d' <- convertExp d
             -- NB: because it's non-recursive let, the freshen must be done after the definition
             b' <- convertFreshenAdd b
             x' <- convertExpQ $ Query cs $ final q
             return $ CE.XLet () b' d' x'
+    -- (Let ann (PatCon ConTuple [PatVariable na, PatVariable nb]) bound:cs)
```

Comment:
remnant cruft?

Reply:
Yeah, I tried to do the translation here first, but it was a bit subtle. I had it working for `let`, but then some possibilities code was breaking.